### PR TITLE
[SPARK-48384][BUILD] Exclude `io.netty:netty-tcnative-boringssl-static` from `zookeeper`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to exclude `io.netty:netty-tcnative-boringssl-static` from `zookeeper`.

### Why are the changes needed?
1.According to:
https://github.com/netty/netty-tcnative/blob/c9b4b6ab62cdbfb4aab6ab3efb8dd7cf73f353ad/boringssl-static/pom.xml#L970-L982
the `io.netty:netty-tcnative-boringssl-static` is composed of:
`io.netty:netty-tcnative-boringssl-static:linux-aarch_64`
`io.netty:netty-tcnative-boringssl-static:linux-x86_64`
`io.netty:netty-tcnative-boringssl-static:osx-aarch_64`
`io.netty:netty-tcnative-boringssl-static:osx-x86_64`
`io.netty:netty-tcnative-boringssl-staticwindows-x86_64`
and our `common/network-common/pom.xml` file already explicitly depends on them.

2.Their versions are different in `dev/deps/spark-deps-hadoop-3-hive-2.3`
image
Let's keep one version to avoid conflicts.

3.In the `pom.xml` file, zookeeper already has other `netty` related exclusions, eg:
https://github.com/apache/spark/blob/master/pom.xml#L1838-L1842
image

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @LuciferYang @dongjoon-hyun

**Comment by panbingkun** [comment]
Although I read inside the contents of `netty-tcnative-boringssl-static-2.0.61.Final.jar` file, there is nothing in it, I  propose we delete it or change it to rely on `netty-tcnative-boringssl-static-2.0.65.Final.jar` to avoid misunderstanding
image

**Comment by LuciferYang** [comment]
> tcnative

If that's the case, both fine for me

**Comment by pan3793** [comment]
cc @mridulm and @hasnain-db as the change is related to SSL dependency

**Comment by hasnain-db** [comment]
Using 2.0.65 seems fine to me (and preferred)

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. I prefer to exclude this. We can add back during RC if this is a required dependency.

However, I'll leave this to the other committers.

**Review by LuciferYang — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master for Spark 4.0. Thanks @panbingkun @dongjoon-hyun @hasnain-db and @pan3793 ~

**Pull request closed without merging**
